### PR TITLE
Remove SPA classes from theme constant selectors

### PR DESCRIPTION
**Pull request by BingoRUS**

### CODE REVIEW

**Review by babich-a — DISMISSED** [review]

**Comment by BingoRUS** [comment]
@jsdmitry @San4es guys, I've changed the logic of `Selector check` test: now we can have any selectors if they *contain* a rule with dx- or ql- prefix. Take a look please.

**Review by San4es — APPROVED** [review]

**Comment by BingoRUS** [comment]
@AlekseyMartynov You're the author of that test (hg commit f8208f69c0cf), review please.

**Review by San4es — DISMISSED** [review]

File: testing/tests/DevExpress.ui/themes.tests.js [hunk 1]
```diff
@@ -45,33 +45,15 @@ require("style-compiler-test-server/known-css-files");
     }
 
     function isGoodSelector(selectorText) {
```
Comment (AlekseyMartynov) [inline testing/tests/DevExpress.ui/themes.tests.js]: Previously this was `false`:
```
isGoodSelector(".dx-button .j123-grid")
```

**Review by AlekseyMartynov — COMMENTED** [review]
How we explain the leading `body`?

File: testing/tests/DevExpress.ui/themes.tests.js (hunk 1 above)
Comment (BingoRUS) [reply testing/tests/DevExpress.ui/themes.tests.js]: It was but it doesn't really make sense for me. I think the main goal for us is not to have any selector that can damage an app outside of our components. Anyway, it's not a big deal to return a class check back. The leading tag is still needed to make some selectors stronger.

**Review by jsdmitry — DISMISSED** [review]

File: testing/tests/DevExpress.ui/themes.tests.js (hunk 1 above)
Comment (AlekseyMartynov) [reply testing/tests/DevExpress.ui/themes.tests.js]: > It was but it doesn't really make sense for me. 

The test case dates back to 2014. If you and peers are confident in the changes, then fine.

> The leading tag is still needed to make some selectors stronger.

Without an example this need is vague. Rules with the leading `body` look odd to me.

Comment (BingoRUS) [reply testing/tests/DevExpress.ui/themes.tests.js]: Fine, I found the problem. [Here](https://github.com/DevExpress/DevExtreme/blob/18_2/styles/widgets/generic/typography.generic.less#L41) we have a selector with class and tag name to color links. If a user uses our typography, classes `.dx-theme-accent-as-text-color`, `.dx-theme-background-color-as-text-color` and `.dx-theme-border-color-as-text-color` will only work if they are not weaker.

**Review by babich-a — APPROVED** [review]

**Review by AlekseyMartynov — APPROVED** [review]
LGTM

**Review by jsdmitry — APPROVED** [review]

**Review by San4es — APPROVED** [review]

**Pull request merged**